mock/mock.go

```diff
@@ -766,7 +766,25 @@ const (
 // AnythingOfTypeArgument contains the type of an argument
 // for use when type checking.  Used in Diff and Assert.
 //
-// Deprecated: this is an implementation detail that must not be used. Use [AnythingOfType] instead.
+// Deprecated: this is an implementation detail that must not be used. Use the [AnythingOfType] constructor instead.
+//
+// If AnythingOfTypeArgument is used as a function return value type you can usually replace
+// the function with a variable. Example:
+//
+//	func anyString() mock.AnythingOfTypeArgument {
+//	    return mock.AnythingOfType("string")
+//	}
+//
+// Can be replaced by:
+//
+//	func anyString() interface{} {
+//	    return mock.IsType("")
+//	}
+//
+// It could even be replaced with a variable:
+//
+//	var anyString = mock.AnythingOfType("string")
+//	var anyString = mock.IsType("") // alternative
```

Comment:
This suggests to replace with a variable, then shows an example which does not do that. Then it suggests again that it can be replaced with a variable and shows two examples.

There are three uses of the type previously, right?

```go
// conversion is replaced by constructor

```


```suggestion
// If AnythingOfTypeArgument is used as a function return value type you can usually replace
// the function with a variable. Example:
//
//	func anyString() mock.AnythingOfTypeArgument {
//	    return mock.AnythingOfType("string")
//	}
//
// Can be replaced by:
//
//	var anyString = mock.AnythingOfType("string")
```